server/server.js

```diff
@@ -252,22 +261,13 @@ app.use('/genotet', function(req, res, next) {
         next();
       } else {
         log.serverLog(result.error);
-        jsonResponse(result, req, res);
+        jsonResponse(/** @type {Object} */(result), req, res);
```

Comment:
You defined user.findUsername to accept a callback: function(user.Error|undefined). That definition is probably conflicting with this one.